ext/storage_sources/s3_store/test/test_s3_connection.cpp

```diff
@@ -325,6 +327,54 @@ TestObjectExists(const Aws::S3Crt::ClientConfiguration &config)
     return (TEST_SUCCESS);
 }
 
+/*
+ * TestBadBucket --
+ *     Unit test to check if connection to a non-existing bucket fails gracefully.
+ */
+int
+TestBadBucket(const Aws::S3Crt::ClientConfiguration &config)
+{
+    int ret = TEST_FAILURE;
+
+    /* The connection object instantitation should not succeed. */
+    try {
+        S3Connection conn(config, "BadBucket", TestDefaults::objPrefix);
+        (void)conn;
+        std::cerr << "TestBadBucket: Failed to generate exception for the bad bucket." << std::endl;
```

Comment:
Should we fail the test here?

Reply:
We do fail here. ```ret``` defaults to ```TEST_FAILURE```, and if the exception is not thrown, ret continues to be set to FAILURE, and we return that as exit code from the test.